java/src/main/java/com/adhoc/flightsql/DremioFlightSqlClientDemoApp.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.adhoc.flightsql;
+
+import org.apache.arrow.flight.CallOption;
+import org.apache.arrow.flight.FlightClient;
+import org.apache.arrow.flight.Location;
+import org.apache.arrow.flight.auth2.BasicAuthCredentialWriter;
+import org.apache.arrow.flight.auth2.ClientBearerHeaderHandler;
+import org.apache.arrow.flight.auth2.ClientIncomingAuthHeaderMiddleware;
+import org.apache.arrow.flight.grpc.CredentialCallOption;
+import org.apache.arrow.flight.sql.FlightSqlClient;
+import org.apache.arrow.flight.sql.example.FlightSqlClientDemoApp;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+/**
+ * Dremio's Flight SQL Client Demo CLI Application.
+ */
+public class DremioFlightSqlClientDemoApp extends FlightSqlClientDemoApp {
+
+  public DremioFlightSqlClientDemoApp(BufferAllocator bufferAllocator) {
+    super(bufferAllocator);
+  }
+
+  public static void main(final String[] args) throws Exception {
+    final Options options = new Options();
+
+    options.addOption("host", "host", true, "Host to connect to");
+    options.addOption("port", "port", true, "Port to connect to");
+    options.addOption("username", "username", true, "Auth username");
+    options.addOption("password", "password", true, "Auth password");
+    options.addRequiredOption("command", "command", true, "Method to run");
+
```

Comment:
Let's add support for pat and generic session properties the same way we have it in the main app.

Reply:
Done